### PR TITLE
Remove the Accumulo connector

### LINKED ISSUE #3558: Add support for 'timestamp with time zone' in Accumulo
Trying to do a `CTAS` from a Postgres table that has a field of type `timestamp with time zone` to an Accumulo table threw the following.  The connector author @adamjshook confirms `timestamp with time zone` currently is not supported.

```
Query 20200424_000300_00009_ir7gy failed: Unsupported PrestoType timestamp with time zone
io.prestosql.spi.PrestoException: Unsupported PrestoType timestamp with time zone
	at io.prestosql.plugin.accumulo.model.Field.cleanObject(Field.java:544)
	at io.prestosql.plugin.accumulo.model.Field.<init>(Field.java:66)
	at io.prestosql.plugin.accumulo.model.Field.<init>(Field.java:61)
	at io.prestosql.plugin.accumulo.model.Row.addField(Row.java:82)
	at io.prestosql.plugin.accumulo.io.AccumuloPageSink.appendPage(AccumuloPageSink.java:241)
	at io.prestosql.operator.TableWriterOperator.addInput(TableWriterOperator.java:238)
	[...]
```

### LINKED ISSUE #5842: Change TestingAccumuloServer to not be a Singleton
Relates to #5598 

Singleton is needed because, due to the fixed ports, multiple containers try to spawn at the same time and it fails. If I try a lock approach so only one container runs at a time, one of the test framework classes (distributed/integration smoke test) inexplicable fail because they are unable to connect to Accumulo once a new container is spawned.

### LINKED ISSUE #16437: Flaky accumulo test: TestAccumuloConnectorTest.testInsertDuplicateRows: Table does not exist
https://github.com/trinodb/trino/actions/runs/4364381826/jobs/7633010303

```
2023-03-08T14:29:03.7960139Z [ERROR] Tests run: 477, Failures: 1, Errors: 0, Skipped: 186, Time elapsed: 364.697 s <<< FAILURE! - in TestSuite
2023-03-08T14:29:03.7961341Z [ERROR] io.trino.plugin.accumulo.TestAccumuloConnectorTest.testInsertDuplicateRows  Time elapsed: 0.261 s  <<< FAILURE!
2023-03-08T14:29:03.7962488Z io.trino.testing.QueryFailedException: line 1:1: Table 'accumulo.tpch.test_insert_duplicate' does not exist
2023-03-08T14:29:03.7963187Z 	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:122)
2023-03-08T14:29:03.7964058Z 	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:489)
2023-03-08T14:29:03.7964686Z 	at io.trino.testing.QueryAssertions.assertUpdate(QueryAssertions.java:71)
2023-03-08T14:29:03.7965550Z 	at io.trino.testing.AbstractTestQueryFramework.assertUpdate(AbstractTestQueryFramework.java:401)
2023-03-08T14:29:03.7966267Z 	at io.trino.testing.AbstractTestQueryFramework.assertUpdate(AbstractTestQueryFramework.java:396)
2023-03-08T14:29:03.7967247Z 	at io.trino.plugin.accumulo.TestAccumuloConnectorTest.testInsertDuplicateRows(TestAccumuloConnectorTest.java:195)
2023-03-08T14:29:03.7967989Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-03-08T14:29:03.7968845Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2023-03-08T14:29:03.7969579Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2023-03-08T14:29:03.7970411Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2023-03-08T14:29:03.7971021Z 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
2023-03-08T14:29:03.7971950Z 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
2023-03-08T14:29:03.7972513Z 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
2023-03-08T14:29:03.7973275Z 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
2023-03-08T14:29:03.7973893Z 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
2023-03-08T14:29:03.7974693Z 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
2023-03-08T14:29:03.7975307Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2023-03-08T14:29:03.7976244Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2023-03-08T14:29:03.7976775Z 	at java.base/java.lang.Thread.run(Thread.java:833)
2023-03-08T14:29:03.7977478Z 	Suppressed: java.lang.Exception: SQL: DROP TABLE test_insert_duplicate
2023-03-08T14:29:03.7978063Z 		at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:492)
2023-03-08T14:29:03.7978730Z 		... 17 more
2023-03-08T14:29:03.7979340Z Caused by: io.trino.spi.TrinoException: line 1:1: Table 'accumulo.tpch.test_insert_duplicate' does not exist
2023-03-08T14:29:03.7980197Z 	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:48)
2023-03-08T14:29:03.7980868Z 	at io.trino.sql.analyzer.SemanticExceptions.semanticException(SemanticExceptions.java:43)
2023-03-08T14:29:03.7981674Z 	at io.trino.execution.DropTableTask.execute(DropTableTask.java:82)
2023-03-08T14:29:03.7982228Z 	at io.trino.execution.DropTableTask.execute(DropTableTask.java:37)
2023-03-08T14:29:03.7983017Z 	at io.trino.execution.DataDefinitionExecution.start(DataDefinitionExecution.java:145)
2023-03-08T14:29:03.7983633Z 	at io.trino.execution.SqlQueryManager.createQuery(SqlQueryManager.java:249)
2023-03-08T14:29:03.7984904Z 	at io.trino.dispatcher.LocalDispatchQuery.lambda$startExecution$7(LocalDispatchQuery.java:143)
2023-03-08T14:29:03.7985527Z 	at io.trino.$gen.Trino_testversion____20230308_142306_71.run(Unknown Source)
2023-03-08T14:29:03.7986322Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
2023-03-08T14:29:03.7986961Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
2023-03-08T14:29:03.7987676Z 	at java.base/java.lang.Thread.run(Thread.java:833)
```

### LINKED ISSUE #20965: Flaky TestIndexer.testMutationIndexWithVisibilities in Accumulo connector
```
Error:  io.trino.plugin.accumulo.index.TestIndexer.testMutationIndexWithVisibilities -- Time elapsed: 5.548 s <<< FAILURE!
org.opentest4j.AssertionFailedError: 

expected: "1"
 but was: "2"
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at io.trino.plugin.accumulo.index.TestIndexer.assertKeyValuePair(TestIndexer.java:336)
	at io.trino.plugin.accumulo.index.TestIndexer.testMutationIndexWithVisibilities(TestIndexer.java:309)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
```
https://github.com/trinodb/trino/actions/runs/8179694894/job/22366292735?pr=17909

### LINKED ISSUE #23792: Remove the Accumulo connetor
The community of maintainers and contributors expressed interest in removing the connector.

### PR DESCRIPTION
**Pull request by mosabua**
## Description

The community expressed interest in removing this connector. More details in the linked issue.

## Additional context and related issues

Fixes #23792
Fixes #20965
Fixes #16437
Fixes #5842
Fixes #3558

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
## Accumulo connector

* Remove the Accumulo connector. ({issue}`23792`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
giphy

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -333,7 +333,6 @@ jobs:
         id: tests
         run: |
           $MAVEN test ${MAVEN_TEST} -pl '
-            !:trino-accumulo,
```
Comment (ebyhr) [inline .github/workflows/ci.yml:336]: `set-session.md` still refers to Accumulo connector.

**Review by ebyhr — COMMENTED** [review]
Though I am not against this removal, this connector is unfortunately used in Trino: The Definitive Guide.

**Comment by mosabua** [comment]
> Though I am not against this removal, this connector is unfortunately used in Trino: The Definitive Guide.

**Comment by mosabua** [comment]
> Though I am not against this removal, this connector is unfortunately used in Trino: The Definitive Guide.

Believe me .. I know about this ;-) .. but honestly I dont think that should stop us. Let me add that info to the issue

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mosabua) [reply .github/workflows/ci.yml:336]: I know .. hence its still a draft .. i will add another commit that changes that whole section.

**Comment by wendigo** [comment]
@ebyhr good point but we should just amend the book

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mosabua) [reply .github/workflows/ci.yml:336]: Updated that example now to use Iceberg.

**Comment by wendigo** [comment]
Can we merge it @martint @electrum @dain ?

**Comment by mosabua** [comment]
I think we should mention it in the contributor and maintainer calls this week and I have to do some more research .. at this stage for the 463 release might be realistic

**Review by raunaqmorarka — APPROVED** [review]

**Comment by mosabua** [comment]
Lets wait over the weekend and into next week for more feedback but I think we can probably merge this for 464.

**Review by ebyhr — APPROVED** [review]

**Comment by mosabua** [comment]
Failures are false alarms  ..

**Pull request merged**